src/CoreWCF.NetTcp/src/CoreWCF/Channels/Framing/ServerSingletonConnectionReaderMiddleware.cs

```diff
@@ -48,7 +48,8 @@ private async Task EnsureChannelAsync(FramingConnection connection)
         {
             if (_replyChannel == null)
             {
-                using (await _lock.TakeLockAsync())
+                var releaser = await _lock.TakeLockAsync();
+                try
```

Comment:
You can make usings async like this:
```c#
await using (await _lock.TakeLockAsync())
{
    // ....
}
```
This will make the DisposeAsync call be awaited like you did manually.

Reply:
CoreWCF is targeting .net standard 2.0 [which defaults ot a C# language version of 7.3](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version#defaults), await usings are a feature of C# 8.0. 
Therefore I had to manually write a try-finally pattern.

Alternatively CoreWCF could activate `<LangVersion>8.0</LangVersion>` in the Directory.Build.props. IIRC various syntactic sugar features like `await using` should work.

Reply:
Can you bump the lang version up to 8.0? There are some 8.0 features which don't work because there needs to be some runtime classes available, but they will fail to compile if we use them so it's safe.